### ISSUE TITLE
warning: integer overflow in expression of type ‘long int’ results in ‘0’

### ISSUE THREAD

**Issue by df7cb** [okbob/pspg#231]
This warning on i386 (32-bit) Debian looks relevant:

```
16:02:22 src/table.c: In function ‘cut_numeric_value’:
16:02:22 src/table.c:1788:30: warning: integer overflow in expression of type ‘long int’ results in ‘0’ [-Woverflow]
16:02:22      mp = 1024l * 1024 * 1024 * 1024;
16:02:22                               ^
16:02:22 src/table.c:1800:30: warning: integer overflow in expression of type ‘long int’ results in ‘0’ [-Woverflow]
16:02:22      mp = 1024l * 1024 * 1024 * 1024;
16:02:22                               ^
```

sizeof(long) is only 4 bytes on 32-bit architectures.

**Comment by okbob** [comment]
should be fixed by e71e48fb6370ec3562cefa69685ccf30d4711cde, please check

**Comment by okbob** [comment]
fixed in 5.8.0
